.github/workflows/build-and-publish-beta-chart.yml

```diff
@@ -0,0 +1,157 @@
+          sed -i ''$VERSION_LINE's/.*/version: ${{ env.NEW_VERSION }}/' charts/cloudzero-agent/Chart.yaml
+
+      # Step 5: Package and Commit Chart
+      - name: Package Chart
+        run: helm package charts/cloudzero-agent/ --destination .deploy
+
+      - name: Get Main Changelog Beginning Hash
+        run: |
+          MAIN_PREV_COMMIT_HASH=$(git rev-parse HEAD)
+          echo "MAIN_PREV_COMMIT_HASH=${MAIN_PREV_COMMIT_HASH}" >> $GITHUB_ENV
+
+      - name: Commit updated Chart.yaml
+        run: |
+          git add .
+          git commit -m "Update Chart.yaml to version ${{ env.NEW_VERSION }}"
+          git push origin main
+          COMMIT_HASH=$(git rev-parse HEAD)
+          echo "COMMIT_HASH=${COMMIT_HASH}" >> $GITHUB_ENV
+        continue-on-error: true
+
+      # Step 6: Generate Change Log
+      - name: Generate Change Log
+        id: get_changes
+        run: |
+          FROM=${{ env.MAIN_PREV_COMMIT_HASH }}
+          TO=$(git rev-parse --short HEAD)
+          CHANGES=$(git log ${FROM}..${TO} --oneline)
+          echo "::set-output name=changes::${CHANGES}"
+
+      # Step 7: Handle Artifacts and Update Pages
+      - name: Upload Chart as Artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: agent-chart
+          path: .deploy/cloudzero-agent-${{ env.NEW_VERSION }}.tgz
+
+      - name: Checkout GH Pages
+        run: |
+          git checkout -f gh-pages
+
+      - name: Move release Tarball
+        run: |
+          cp .deploy/cloudzero-agent-${{ env.NEW_VERSION }}.tgz ./
+          rm -fr .deploy
+
+      - name: Update Index
+        run: helm repo index --url https://cloudzero.github.io/cloudzero-charts .
+
+      - name: Save Index in GH Pages
+        run: |
+          # copy the new chart and index.yaml
+          git add cloudzero-agent-${{ env.NEW_VERSION }}.tgz index.yaml 
+          git commit -m "Updating ${{ env.NEW_VERSION }} Index"
+          git push origin gh-pages
+        continue-on-error: true
+
+      - name: Update Docs for GH Pages
+        run: |
+          # cleanup garbage files
+          rm -fr .deploy charts/cloudzero-agent/charts
+          git reset --hard 
+          # now checkout docs from main
+          git checkout main -- charts/cloudzero-agent/docs charts/cloudzero-agent/README.md README.md
+          git add README.md charts/cloudzero-agent/docs charts/cloudzero-agent/README.md
+          git commit -m "Update docs for ${{ env.NEW_VERSION }}"
+          git push origin gh-pages
+        continue-on-error: true
+
+      # Step 8: Create GitHub Release
+      - name: Create Release
+        uses: softprops/action-gh-release@v2
+        with:
+          name: ${{ env.NEW_VERSION }}
+          tag_name: ${{ env.NEW_VERSION }}
+          files: cloudzero-agent-${{ env.NEW_VERSION }}.tgz
+          make_latest: ${{ github.event.inputs.release_type != 'beta' }}
```

Comment:
For anyone confused by this line:

If github.event.inputs.release_type is 'beta', the condition github.event.inputs.release_type != 'beta' evaluates to false, so make_latest is set to false. We don't want beta tags set to latest.